python/tank_vendor/shotgun_deploy/resolver.py

```diff
@@ -65,12 +65,104 @@ def resolve_configuration(
         raise NotImplementedError
 
 
+class NoShotgunFallbackConfigurationResolver(ConfigurationResolver):
```

Comment:
very minor thought: perhaps this should just be `BaseConfigurationResolver`? Or maybe just a function `resolve_base_configuration()`?